lib/text.ts

```diff
@@ -0,0 +1,96 @@
+import * as changeCase from "change-case";
+import { convert as slugify } from "url-slug";
+
+export function countWords(text: string): number {
+  // Split the text by any whitespace (spaces, newlines, tabs, etc.)
+  const words = text.split(/\s+/);
+  // Filter out any empty strings in case there are extra spaces or newlines
+  const filteredWords = words.filter(word => word.length > 0);
+  return filteredWords.length;
+}
+
+export function countCharacters(text: string): number {
+  /** Unicode-proof way to get string length in codepoints (in characters)
+   For example:
+   const text = '👨‍👩‍👧‍👦';
+   text.length === 11;
+   [...new Intl.Segmenter().segment(text)].length === 1;
+   */
+  return [...new Intl.Segmenter().segment(text)].length;
+}
+
+export function countLines(text: string): number {
+  return text.split("\n").length;
+}
+
+export function countBytes(text: string): number {
+  return new Blob([text]).size;
```

Comment:
One possible approach is to use `new TextEncoder().encode(text).length`.
I think it is easy to understand that the text is converted to UTF-8.
What do you think? 😃